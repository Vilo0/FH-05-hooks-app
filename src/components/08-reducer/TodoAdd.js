import { useForm } from "../../hooks/useForm";

const TodoAdd = ({ handleAddTodo }) => {

    const [{ description }, handleInputChange, reset] = useForm({
        description: ''
    });

    const handleSubmit = (e) => {
        e.preventDefault();

        if(description.trim().length <= 1)
            return;

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false,
        }

        handleAddTodo(newTodo);
        reset();
    }

    return (
        <>
            <h4>Agregar Todo</h4>
            <hr />
            
            <form onSubmit={ handleSubmit }>
                
                <input 
                    type="text" 
                    name="description"
                    placeholder="Aprender ..."
                    autoComplete="false"
                    className="form-control"
                    onChange={ handleInputChange }
                    value={ description }
                />

                <button
                    type="submit" 
                    className="btn btn-outline-primary mt-2 btn-block"
                >
                    Agregar
                </button>

            </form>
        </>
    )
}

export default TodoAdd;
