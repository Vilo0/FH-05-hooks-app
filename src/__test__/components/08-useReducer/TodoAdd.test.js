import { shallow } from "enzyme";
import { act } from "react-dom/test-utils";
import TodoAdd from "../../../components/08-reducer/TodoAdd";

describe('Pruebas en <TodoAdd />', () => {
    const handleAddTodo = jest.fn();
    const wrapper = shallow(
        <TodoAdd 
            handleAddTodo={handleAddTodo}
        />
    );

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('NO debe de llamar handleAddTodo', () => {
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({ preventDefault(){} });

        expect(handleAddTodo).not.toHaveBeenCalled();
    });

    test('debe de llamar la funciÃ³n handleAddTodo', () => {
        const value = 'Aprender React';
        wrapper.find('input').simulate('change', {
            target: {
                value,
                name: 'description'
            },
        });

        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({ preventDefault(){} });

        expect(handleAddTodo).toHaveBeenCalled();
        expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
        expect(handleAddTodo).toHaveBeenCalledWith({
            id: expect.any(Number),
            desc: value,
            done: false,
        });

        expect(wrapper.find('input').prop('value')).toBe('');
    });
});