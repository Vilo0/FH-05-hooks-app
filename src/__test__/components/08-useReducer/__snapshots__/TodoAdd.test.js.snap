// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoAdd /> NO debe de llamar addTODO 1`] = `
<Fragment>
  <h4>
    Agregar Todo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="false"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender ..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-2 btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;

exports[`Pruebas en <TodoAdd /> NO debe de llamar handleAddTodo 1`] = `
<Fragment>
  <h4>
    Agregar Todo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="false"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender ..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-2 btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;

exports[`Pruebas en <TodoAdd /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h4>
    Agregar Todo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="false"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender ..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-2 btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
